internal/controller/policies.go

```diff
@@ -30,6 +32,52 @@ func (p PolicyTargetKey) String() string {
 	return p.NsName.String() + "/" + p.GroupKind.String()
 }
 
+func BackendTrafficPolicyPredicateFunc(channel chan event.GenericEvent) predicate.Predicate {
+	return predicate.Funcs{
+		GenericFunc: func(e event.GenericEvent) bool {
+			return false
+		},
+		DeleteFunc: func(e event.DeleteEvent) bool {
+			return true
+		},
+		CreateFunc: func(e event.CreateEvent) bool {
+			return true
+		},
+		UpdateFunc: func(e event.UpdateEvent) bool {
+			oldObj, ok := e.ObjectOld.(*v1alpha1.BackendTrafficPolicy)
+			newObj, ok2 := e.ObjectNew.(*v1alpha1.BackendTrafficPolicy)
+			if !ok || !ok2 {
+				return false
+			}
+			oldRefs := oldObj.Spec.TargetRefs
+			newRefs := newObj.Spec.TargetRefs
+
+			oldRefMap := make(map[string]v1alpha1.BackendPolicyTargetReferenceWithSectionName)
+			for _, ref := range oldRefs {
+				key := fmt.Sprintf("%s/%s/%s", ref.Group, ref.Kind, ref.Name)
+				oldRefMap[key] = ref
+			}
+
+			for _, ref := range newRefs {
+				key := fmt.Sprintf("%s/%s/%s", ref.Group, ref.Kind, ref.Name)
+				delete(oldRefMap, key)
+			}
+			if len(oldRefMap) > 0 {
+				targetRefs := make([]v1alpha1.BackendPolicyTargetReferenceWithSectionName, 0, len(oldRefs))
+				for _, ref := range oldRefMap {
+					targetRefs = append(targetRefs, ref)
+				}
+				dump := oldObj.DeepCopy()
+				dump.Spec.TargetRefs = targetRefs
+				channel <- event.GenericEvent{
+					Object: dump,
+				}
+			}
```

Comment:
For HTTPRoute reconciler, only need to return refs which's kind is "HTTPRoute", for Ingress reconciler, only need to return refs which's kind is "Ingress". Do we need to filter refs by Kind here?

Reply:
Isn't it more appropriate for the consumer of the event to decide the filtering criteria? 
What do you think?